presto-main/src/main/java/com/facebook/presto/sql/gen/JoinCompiler.java

```diff
@@ -243,6 +254,22 @@ private static void generateConstructor(ClassDefinition classDefinition,
                 .ifFalse(thisVariable.setField(
                         hashChannelField,
                         constantNull(hashChannelField.getType()))));
+
+        constructor.comment("check if join filter function is passed only if field is generated");
+        if (joinFilterFunctionFieldOptional.isPresent()) {
+            constructor.append(invokeStatic(Preconditions.class, "checkArgument", void.class, ImmutableList.of(boolean.class, Object.class),
```

Comment:
Yeah - somewhat.
We generate the constructor. But the call is elsewhere. If this check is not present the code will fail when generated `PagesHashStrategy.applyFilterFunction` is called.
I can drop it if you like. Please confirm this or that way.